From 55b8254f9f18eb68ddada3ad6aaefdbd64c57268 Mon Sep 17 00:00:00 2001
From: Marko Kallio <mkallio@marvell.com>
Date: Fri, 21 Sep 2018 12:44:28 +0530
Subject: [PATCH 0203/1051] soc: octeontx2: reset all mbox devid in up handler

Right now, reset happens for only devid `0` and
mbox up msgs for other devids are not flushed.

Spotted while debugging [1]:

[1]:
`````````
[  113.907679] octeontx2-af 0002:01:00.0: notification to pf 1 failed
`````````

Signed-off-by: Marko Kallio <mkallio@marvell.com>
Signed-off-by: Santosh Shukla <sshukla@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu.c b/drivers/soc/marvell/octeontx2/rvu.c
index 7c129f75732c..ed368f36327a 100644
--- a/drivers/soc/marvell/octeontx2/rvu.c
+++ b/drivers/soc/marvell/octeontx2/rvu.c
@@ -1582,7 +1582,7 @@ static void __rvu_mbox_up_handler(struct rvu_work *mwork, int type)
 		mdev->msgs_acked++;
 	}
 
-	otx2_mbox_reset(mbox, 0);
+	otx2_mbox_reset(mbox, devid);
 }
 
 static inline void rvu_afpf_mbox_up_handler(struct work_struct *work)
-- 
2.17.1

